lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1417 @@
+    if ($mirrors.Length -gt 1){
+        Set-IniFileValue "extra-index-url" "global" ($mirrors[1..$mirrors.Length] -Join " ") $pipIni
+    }
+    Set-IniFileValue "trusted-host" "install" ($hosts -Join " ") $pipIni
+}
+
+
+function Get-HypervADUser {
+    $adUsername = Get-JujuCharmConfig -scope 'ad-user-name'
+    if (!$adUsername) {
+        $adUsername = "hyper-v-user"
+    }
+    return $adUsername
+}
+
+
+function Set-DevStackRelationParams {
+    Param(
+        [HashTable]$RelationParams
+    )
+
+    $rids = Get-JujuRelationIds -Relation "devstack"
+    foreach ($rid in $rids) {
+        try {
+            Set-JujuRelation -Settings $RelationParams -RelationId $rid
+        } catch {
+            Write-JujuError "Failed to set DevStack relation parameters."
+        }
+    }
+}
+
+
+function Import-CloudbaseCert {
+    $crt = Join-Path $FILES_DIR "Cloudbase_signing.cer"
+    if (!(Test-Path $crt)){
+        return $false
+    }
+    Import-Certificate $crt -StoreLocation LocalMachine -StoreName TrustedPublisher
+}
+
+
+function Set-AD2DevstackCreds {
+    $adUserCred = @{
+        'domain'   = $adCtx["domainName"];
+        'username' = $adCtx['adcredentials'][0]['username'];
+        'password' = $adCtx['adcredentials'][0]['password']
+    }
+    $relationParams = @{'ad_credentials' = (Get-MarshaledObject $adUserCred);}
+    Set-DevStackRelationParams $relationParams
+}
+
+
+function Join-ADDomain {
+    $adCtx = Get-ActiveDirectoryContext
+    if (Confirm-IsInDomain $adCtx["domainName"]) {
+        # Add AD user to local Administrators group
+            Start-ExecuteWithRetry {
+                Grant-PrivilegesOnDomainUser -Username $adCtx['adcredentials'][0]['username']
+            }
+            Enable-LiveMigration
+    }
+    else {
+        Start-JoinDomain
+    }
+}
+
+
+function Enable-LiveMigration {
+    Write-JujuLog "Enabling live migration"
+    Enable-VMMigration
+    $name = Get-MainNetadapter
+    $netAddresses = Get-NetIPAddress -InterfaceAlias $name -AddressFamily IPv4
+    foreach($netAddress in $netAddresses) {
+        $prefixLength = $netAddress.PrefixLength
+        $netmask = ConvertTo-Mask -MaskLength $prefixLength
+        $networkAddress = Get-NetworkAddress -IPAddress $netAddress.IPAddress -SubnetMask $netmask
+        $migrationNet = Get-VMMigrationNetwork | Where-Object { $_.Subnet -eq "$networkAddress/$prefixLength" }
+        if (!$migrationNet) {
+            Add-VMMigrationNetwork -Subnet "$networkAddress/$prefixLength" -Confirm:$false
+        }
+    }
+}
+
+
+# HOOKS FUNCTIONS
+
+function Invoke-InstallHook {
+    # Set machine to use high performance settings.
+    try {
+        Set-PowerProfile -PowerProfile Performance
+    } catch {
+        # No need to error out the hook if this fails.
+        Write-JujuWarning "Failed to set power scheme."
+    }
+
+    Start-TimeResync
+
+    # Set Administrator user password
+    $admin_user = "Administrator"
```

Comment:
Why exactly do you need to set the default Administrator password ?

And btw, you can get the default `Administrator` name with the function `Get-AdministratorAccount` from `JujuWindowsUtils` and it works on every internationalized Windows version as well. I would recommend using it instead of this hardcoded value.